src/main/java/com/featureprobe/api/repository/SegmentRepository.java

```diff
@@ -23,4 +25,18 @@ public interface SegmentRepository extends JpaRepository<Segment, Long>, JpaSpec
     @Query(value = "SELECT count(id) FROM segment WHERE project_key = ?1 AND name = ?2", nativeQuery = true)
     int countByNameIncludeDeleted(String projectKey, String name);
 
+    @Query(
+            value = "SELECT * FROM segment WHERE ( if(?2 != '', name like concat('%',?2,'%'), 1=1) "
+                    + "OR if(?2 != '', `key` like concat('%',?2,'%'), 1=1) "
+                    + "OR if(?2 != '', description like concat('%',?2,'%'), 1=1) )"
+                    + "AND project_key = ?1 "
+                    + "ORDER BY created_time DESC",
+            countQuery = "SELECT count(*) FROM segment WHERE ( if(?2 != null, name like concat('%',?2,'%'), 1=1) "
+                    + "OR if(?2 != '', `key` like concat('%',?2,'%'), 1=1) "
+                    + "OR if(?2 != '', description like concat('%',?2,'%'), 1=1) )"
+                    + "AND project_key = ?1 ",
+            nativeQuery = true
+    )
```

Comment:
Non-standard generic sql, like `OR if`, `concat` .. recommended to use `Specification`

Ref: https://pavankjadda.medium.com/search-data-across-multiple-columns-using-spring-data-jpa-fc0a24799991

Reply:
entity used @Where(clause = "deleted = 0") ，so need native Query